src/state/slices/opportunitiesSlice/resolvers/utils.ts

```diff
@@ -0,0 +1,43 @@
+import type { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import pipe from 'lodash/flow'
+
+import {
+  DefiProviderToMetadataResolverByDeFiType,
+  DefiProviderToUserDataResolverByDeFiType,
+} from '.'
+
+// "Give me the resolvers for a given DeFi provider"
+export const getDefiProviderMetadataResolvers = (defiProvider: DefiProvider) =>
+  DefiProviderToMetadataResolverByDeFiType[defiProvider]
+// "Give me the resolvers for a given DeFi type"
+export const getDefiTypeMetadataResolvers = (
+  defiType: DefiType,
+  resolversByType: ReturnType<typeof getDefiProviderUserDataResolvers>,
```

Comment:
I think this type is wrong.

```suggestion
  resolversByType: ReturnType<typeof getDefiProviderMetadataResolvers>,
```

Reply:
I'd also prefer to see `getDefiProviderMetadataResolvers`'s (and `getDefiProviderUserDataResolvers`) return type defined explicitly, and then used here.